src/shared/persistence/scope-base.ts

```diff
@@ -16,6 +16,9 @@ export abstract class ScopeBase<T extends AnyEntity> {
     protected abstract get entityName(): EntityName<T>;
 
     public setScopeWhereOperator(operator: ScopeOperator): this {
+        if (this.scopeWhereOperator) {
```

Comment:
Maybe extend the message a little indicating that we cannot do nested? It would help a dev to understand that what they're trying won't work